src/lib/libfs.js

```diff
@@ -1044,10 +1072,14 @@ FS.staticInit();
     },
     ftruncate(fd, len) {
       var stream = FS.getStreamChecked(fd);
-      if ((stream.flags & {{{ cDefs.O_ACCMODE }}}) === {{{ cDefs.O_RDONLY}}}) {
+      if (len < 0 || (stream.flags & {{{ cDefs.O_ACCMODE }}}) === {{{ cDefs.O_RDONLY}}}) {
         throw new FS.ErrnoError({{{ cDefs.EINVAL }}});
       }
-      FS.truncate(stream.node, len);
+      FS.truncateChecks(stream.node);
+      FS.streamSetAttr(stream, {
+        size: len,
+        timestamp: Date.now()
+      });
```

Comment:
Same here..  IIRC we found a way to handle this kind of thing were we want to do that same for nodes and streams.